packages/babel-parser/src/parser/expression.js

```diff
@@ -2101,6 +2101,8 @@ export default class ExpressionParser extends LValParser {
     node: N.BodilessFunctionOrMethodBase,
     type: string,
     isMethod?: boolean = false,
+    // eslint-disable-next-line no-unused-vars -- this is used in /plugins/typescript
+    isDeclare?: boolean = false,
```

Comment:
Not a big fan of modifying the base parser for this, but it does seem like it might be complicated to overload or wrap a bunch of functions in the plugin. Curious what the rest of the team thinks.